src/format/strftime.rs

```diff
@@ -416,6 +455,23 @@ impl<'a> Iterator for StrftimeItems<'a> {
 }
 
 impl<'a> StrftimeItems<'a> {
+    fn error<'b>(
+        &mut self,
+        original: &'b str,
+        error_len: &mut usize,
+        ch: Option<char>,
+    ) -> (&'b str, Item<'b>) {
+        if let Some(c) = ch {
+            if self.lossy {
+                *error_len -= c.len_utf8();
+            }
+        }
+        (
+            &original[*error_len..],
+            if self.lossy { Item::Literal(&original[..*error_len]) } else { Item::Error },
+        )
```

Comment:
I think this would be clearer if transposed:

```rust
if !self.lossy {
     return (&original[*error_len..], Item::Error);
}

if let Some(c) = ch {
     *error_len -= c.len_utf8();
}
(&original[*error_len..], Item::Literal(&original[..*error_len]))
```